iSeg/LoaderWidgets.cpp

```diff
@@ -1254,80 +979,169 @@ LoaderColorImages::LoaderColorImages(SlicesHandler* hand3D, eImageType typ, std:
 	, type(typ)
 	, m_filenames(filenames)
 {
-	vbox1 = new Q3VBox(this);
-	hbox2 = new Q3HBox(vbox1);
-	subsect = new QCheckBox(QString("Subsection "), hbox2);
+	map_to_lut = new QCheckBox(QString("Map colors to lookup table"));
+	map_to_lut->setChecked(true);
+
+	subsect = new QCheckBox(QString("Subsection"));
 	subsect->setChecked(false);
-	subsect->show();
-	vbox2 = new Q3VBox(hbox2);
-	hbox3 = new Q3HBox(vbox2);
-	xoffs = new QLabel(QString("x-Offset: "), hbox3);
-	xoffset = new QSpinBox(0, 2000, 1, hbox3);
+
+	auto xoffs = new QLabel(QString("x-Offset: "));
+	xoffset = new QSpinBox(0, 2000, 1, nullptr);
 	xoffset->setValue(0);
-	xoffset->show();
-	yoffs = new QLabel(QString("y-Offset: "), hbox3);
-	yoffset = new QSpinBox(0, 2000, 1, hbox3);
-	yoffset->show();
+
+	auto yoffs = new QLabel(QString("y-Offset: "));
+	yoffset = new QSpinBox(0, 2000, 1, nullptr);
 	xoffset->setValue(0);
-	hbox3->show();
-	hbox4 = new Q3HBox(vbox2);
-	xl = new QLabel(QString("x-Length: "), hbox4);
-	xlength = new QSpinBox(0, 2000, 1, hbox4);
-	xlength->show();
-	xlength->setValue(256);
-	yl = new QLabel(QString("y-Length: "), hbox4);
-	ylength = new QSpinBox(0, 2000, 1, hbox4);
-	ylength->show();
-	ylength->setValue(256);
-	hbox4->show();
-	vbox2->show();
-	hbox2->show();
-	hbox6 = new Q3HBox(vbox1);
-	loadFile = new QPushButton("Open", hbox6);
-	cancelBut = new QPushButton("Cancel", hbox6);
-	hbox6->show();
 
-	vbox1->show();
-	setSizePolicy(QSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed));
+	auto xl = new QLabel(QString("x-Length: "));
+	xlength = new QSpinBox(0, 2000, 1, nullptr);
+	xlength->setValue(256);
 
-	vbox1->setFixedSize(vbox1->sizeHint());
+	auto yl = new QLabel(QString("y-Length: "));
+	ylength = new QSpinBox(0, 2000, 1, nullptr);
+	ylength->setValue(256);
 
-	subsect_toggled(subsect->isChecked());
+	auto subsect_layout = new QGridLayout(2,4);
```

Comment:
you can also pass the layout in the widget constructor (the layout acts as parent if I'm not wrong). or use the qt designer which moves all this GUI stuff in XML-based files.